src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -435,24 +517,29 @@ protected virtual async Task<Document> PostProcessChangesAsync(Document document
         /// <param name="equivalenceKey">Optional value used to determine the equivalence of the <see cref="CodeAction"/> with other <see cref="CodeAction"/>s. See <see cref="CodeAction.EquivalenceKey"/>.</param>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static CodeAction Create(string title, Func<CancellationToken, Task<Document>> createChangedDocument, string? equivalenceKey)
+            => Create(title, (_, c) => createChangedDocument(c), equivalenceKey);
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
>ditorBrowsableState.Never)] [](http://example.com/codeflow?start=26&length=27)

why EditorBrowsableState.Never?

Reply:
it allows for binary compat (The older api is there), but it guides people to just use thsi API where they are passed the progress object.  They don't need to use it, but they see that htey have it, and thus can be primed to use it if they feel it's sensible.